sdks/python/apache_beam/utils/dependency.py

```diff
@@ -435,7 +435,13 @@ def _stage_dataflow_sdk_tarball(sdk_remote_location, staged_path, temp_dir):
     _dependency_file_copy(sdk_remote_location, staged_path)
   elif sdk_remote_location == 'pypi':
     logging.info('Staging the SDK tarball from PyPI to %s', staged_path)
-    _dependency_file_copy(_download_pypi_sdk_package(temp_dir), staged_path)
+    import pkg_resources as pkg
```

Comment:
It would make sense to push down this check to `_download_pypi_sdk_package`. This will remove the requirement for importing the same package here again, and you can raise the same RuntimeError there.